src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -408,8 +415,15 @@ public function display()
         $this->overwrite(preg_replace_callback("{%([a-z\-_]+)(?:\:([^%]+))?%}i", function ($matches) {
             if ($formatter = $this::getPlaceholderFormatterDefinition($matches[1])) {
                 $text = call_user_func($formatter, $this, $this->output);
+                if ($this->adjustBarWidthToWindowWidth($text)) {
```

Comment:
The `$text` variable here is not the whole line, but just different fragments of it. The whole line is the return value of the `preg_replace_callback` function.